src/Components/Endpoints/test/EndpointHtmlRendererTest.cs

```diff
@@ -46,6 +46,21 @@ public EndpointHtmlRendererTest()
         renderer = GetEndpointHtmlRenderer();
     }
 
+    [Fact]
+    public async Task DoesNotRenderChildAfterRendererStopped()
+    {
+        renderer.SignalRendererToFinishRendering();
+
+        var httpContext = GetHttpContext();
+        var writer = new StringWriter();
+
+        var result = await renderer.PrerenderComponentAsync(httpContext, typeof(SimpleComponent), null, ParameterView.Empty);
+        await renderer.Dispatcher.InvokeAsync(() => result.WriteTo(writer, HtmlEncoder.Default));
+        var content = writer.ToString();
+
+        Assert.DoesNotContain("Hello from SimpleComponent", content);
+    }
+
```

Comment:
The more interesting case here is where you have something like

`override OnInitializedAsync() { await taskCompletionSource.Task; _state = true; }` with something that changes the rendered stuff based on `_state` and then you do `renderer.SignalRenderToFinishRendering(); taskCompletionSource.SetResult()` and either check the output or count the number of render batches generated.